accounts-db/src/append_vec.rs

```diff
@@ -1402,29 +1402,6 @@ pub mod tests {
 
                 // upper 7-bits are not 0, so sanitization should fail
                 assert!(!account.sanitize_executable());
-
-                // we can observe crafted value by ref
-                {
-                    let executable_bool: &bool = &account.account_meta.executable;
-                    // Depending on use, *executable_bool can be truthy or falsy due to direct memory manipulation
-                    // assert_eq! thinks *executable_bool is equal to false but the if condition thinks it's not, contradictorily.
-                    assert!(!*executable_bool);
-                    #[cfg(not(target_arch = "aarch64"))]
-                    {
-                        const FALSE: bool = false; // keep clippy happy
-                        if *executable_bool == FALSE {
-                            panic!("This didn't occur if this test passed.");
-                        }
-                    }
-                    assert_eq!(*account.ref_executable_byte(), crafted_executable);
-                }
-
-                // we can NOT observe crafted value by value
-                {
-                    let executable_bool: bool = account.executable();
-                    assert!(!executable_bool);
-                    assert_eq!(account.get_executable_byte(), 0); // Wow, not crafted_executable!
-                }
```

Comment:
if I understand correctly, the `executable_bool` has been corrupted because we forcibly write a byte into a bool. If we can catch it by `sanitize_executable`, I guess we don't need to care about the value 🤔